x/clock/abci_test.go

```diff
@@ -0,0 +1,267 @@
+	s.Require().Equal(expHash[:], result.Checksum)
+	// and
+	info := s.app.AppKeepers.WasmKeeper.GetCodeInfo(s.ctx, 1)
+	s.Require().NotNil(info)
+	s.Require().Equal(expHash[:], info.CodeHash)
+	s.Require().Equal(sender.String(), info.Creator)
+	s.Require().Equal(wasmtypes.DefaultParams().InstantiateDefaultPermission.With(sender), info.InstantiateConfig)
+}
+
+func (s *EndBlockerTestSuite) InstantiateContract(sender string, admin string) string {
+	msgStoreCode := wasmtypes.MsgStoreCodeFixture(func(m *wasmtypes.MsgStoreCode) {
+		m.WASMByteCode = clockContract
+		m.Sender = sender
+	})
+	_, err := s.app.MsgServiceRouter().Handler(msgStoreCode)(s.ctx, msgStoreCode)
+	s.Require().NoError(err)
+
+	msgInstantiate := wasmtypes.MsgInstantiateContractFixture(func(m *wasmtypes.MsgInstantiateContract) {
+		m.Sender = sender
+		m.Admin = admin
+		m.Msg = []byte(`{}`)
+	})
+	resp, err := s.app.MsgServiceRouter().Handler(msgInstantiate)(s.ctx, msgInstantiate)
+	s.Require().NoError(err)
+	var result wasmtypes.MsgInstantiateContractResponse
+	s.Require().NoError(s.app.AppCodec().Unmarshal(resp.Data, &result))
+	contractInfo := s.app.AppKeepers.WasmKeeper.GetContractInfo(s.ctx, sdk.MustAccAddressFromBech32(result.Address))
+	s.Require().Equal(contractInfo.CodeID, uint64(1))
+	s.Require().Equal(contractInfo.Admin, admin)
+	s.Require().Equal(contractInfo.Creator, sender)
+
+	return result.Address
+}
+
+func (s *EndBlockerTestSuite) FundAccount(ctx sdk.Context, addr sdk.AccAddress, amounts sdk.Coins) error {
+	if err := s.app.AppKeepers.BankKeeper.MintCoins(ctx, minttypes.ModuleName, amounts); err != nil {
+		return err
+	}
+
+	return s.app.AppKeepers.BankKeeper.SendCoinsFromModuleToAccount(ctx, minttypes.ModuleName, addr, amounts)
+}
+
+// Register a contract. You must store the contract code before registering.
+func (s *EndBlockerTestSuite) registerContract() string {
+	// Create & fund accounts
+	_, _, sender := testdata.KeyTestPubAddr()
+	_, _, admin := testdata.KeyTestPubAddr()
+	_ = s.FundAccount(s.ctx, sender, sdk.NewCoins(sdk.NewCoin("stake", sdk.NewInt(1_000_000))))
+	_ = s.FundAccount(s.ctx, admin, sdk.NewCoins(sdk.NewCoin("stake", sdk.NewInt(1_000_000))))
+
+	// Instantiate contract
+	contractAddress := s.InstantiateContract(sender.String(), admin.String())
+
+	// Register contract
+	clockKeeper := s.app.AppKeepers.ClockKeeper
+	err := clockKeeper.RegisterContract(s.ctx, admin.String(), contractAddress)
+	s.Require().NoError(err)
+
+	// Assert contract is registered
+	contract, err := clockKeeper.GetClockContract(s.ctx, contractAddress)
+	s.Require().NoError(err)
+	s.Require().Equal(contractAddress, contract.ContractAddress)
+
+	// Increment block height
+	s.ctx = s.ctx.WithBlockHeight(11)
+
+	return contract.ContractAddress
+}
+
+// Test the end blocker. This test registers a contract, executes it with enough gas,
+// too little gas, and also ensures the unjailing process functions.
+func (s *EndBlockerTestSuite) TestEndBlocker() {
```

Comment:
Set the BlockGas (consensusParams) to < the contract executes for in the endblocker and ensure that goes through fine as well

Reply:
Line 156 updates the gas limit to 65,000 which is not enough gas for the contract. Is that what you mean?

Reply:
I was referencing `s.ctx = s.ctx.WithBlockGasMeter(sdk.NewGasMeter(1_000_000))` instead, which is the max gas that can be used. But seeing no issues with this after I put it into test, so does not apply :D